interfaces/cython/cantera/examples/thermo/equivalenceRatio.py

```diff
@@ -2,52 +2,115 @@
+Z = gas.mixture_fraction(fuel="CH4", oxidizer="O2:0.21,N2:0.79", element="Bilger")
+Z = gas.mixture_fraction(fuel="CH4", oxidizer="O2:0.21,N2:0.79", element="C")
+
+# Since the fuel in this example is pure methane and the oxidizer is air,
+# the mixture fraction is the same as the mass fraction of methane in the mixture
+print("mass fraction of CH4 = {:1.3f}".format(gas["CH4"].Y[0]))
 
-# The mixture fraction is kg fuel / (kg fuel + kg oxidizer). Since the fuel in
-# this example is pure methane and the oxidizer is air, the mixture fraction
-# is the same as the mass fraction of methane in the mixture
+# To set a mixture according to the mixture fraction, the following function
+# can be used. In this example, the final fuel/oxidizer mixture
+# contains 5.5 mass-% CH4:
+Z = gas.set_mixture_fraction(0.055, fuel="CH4", oxidizer="O2:0.21,N2:0.79")
 print("mass fraction of CH4 = {:1.3f}".format(gas["CH4"].Y[0]))
 
 # Mixture fraction and equivalence ratio are invariant to the reaction progress.
 # For example, they stay constant if the mixture composition changes to the burnt
-# state
+# state or for any intermediate state
+fuel = "CH4:1"
+oxidizer = "O2:0.21,N2:0.79"
+gas.set_equivalence_ratio(1, fuel, oxidizer)
 gas.equilibrate('HP')
 phi_burnt = gas.equivalence_ratio(fuel, oxidizer)
 Z_burnt = gas.mixture_fraction(fuel, oxidizer)
 print("phi(burnt) = {:1.3f}".format(phi_burnt))
 print("Z(burnt) = {:1.3f}".format(Z_burnt))
+
+# If fuel and oxidizer compositions are specified consistently, then
+# equivalence_ratio and set_equivalence_ratio are consistent as well, as
+# shown in the following example
+gas.set_equivalence_ratio(2.5,fuel="CH4:1,O2:0.01,CO:0.05,N2:0.1",
+        oxidizer="O2:0.2,N2:0.8,CO2:0.05,CH4:0.01")
+
+phi = gas.equivalence_ratio(fuel="CH4:1,O2:0.01,CO:0.05,N2:0.1",
+        oxidizer="O2:0.2,N2:0.8,CO2:0.05,CH4:0.01")
+print("phi = {:1.3f}".format(phi)) # prints 2.5
+
+# Without specifying the fuel and oxidizer compositions, it is assumed that
+# all C, H and S atoms come from the fuel and all O atoms from the oxidizer,
+# which is not true for this example. Therefore, the following call gives a
+# different equivalence ratio based on that assumption
+phi = gas.equivalence_ratio()
+print("phi = {:1.3f}".format(phi))
+
+# After computing the mixture composition for a certain equivalence ratio given
+# a fuel and mixture composition, the mixture can optionally be diluted. The
+# following function will first create a mixture with equivalence ratio 2 from pure
+# hydrogen and oxygen and then dilute it with H2O. In this example, the final mixture
+# consists of 30 mol-% H2O and 70 mol-% of the H2/O2 mixture at phi=2
+gas.set_equivalence_ratio(2.0, "H2:1", "O2:1", fraction={"diluent":0.3}, diluent="H2O")
+print("mole fraction of H2O = {:1.3f}".format(gas["H2O"].X[0])) # final mixture contains 30 mol-% H2O
+print("ratio of H2/O2: {:1.3f}".format(gas["H2"].X[0]/gas["O2"].X[0])) # according to phi=2
```

Comment:
With our most recent Python requirements, you can start using f-strings (which are simpler), e.g.
```python
print(f"mole fraction of H2O = {gas['H2O'].X[0]:1.3f}")
```